src/scenes/characters/character.gd

```diff
@@ -17,9 +17,26 @@ func hover_unfocus() -> void:
 func interact() -> void:
     if (!GameState.is_in_interaction()):
         GameState.set_is_in_dialogue(true);
-        var dialog = Dialogic.start(npc + str(dialog_index));
+        quest_status();
+        var index = Dialogic.get_variable(npc + '_index');
+        var dialog = Dialogic.start(npc + str(index));
         dialog.connect("timeline_end", self, "dialog_ended");
+        dialog.connect("dialogic_signal", self, "dialogic_signal_event");
         add_child(dialog);
 
+func quest_status() -> void:
```

Comment:
This implementation is fine for this application. If you wanted to make the quest system more modular you would delegate this responsibility to the quest subclass itself, e.g.:
```gd
class_name FoobarQuest

func get_quest_step() -> QuestStep: # Note: I don't think you can annotate return type with an enum in Godot
  if (Inventory.is_item_picked_up(quest_item)):
    return QuestStep.COMPLETE;
```

Then in your character:
```gd
func interact() -> void:
  if (quest.get_quest_step() == QuestStep.COMPLETE):
    Dialogic.start('quest_complete_dialogue');
```

Not required for this project. Just wanted to demonstrate what a state-based solution would look like, and how I would delegate responsibilities.

Reply:
ahh thanks for the demonstration! This would definitely make the code more clear and readable.